Add Vended Credentials mechanism for Azure (ADLS)

<!-- (At least include the following, feel free to add if you have more content) -->

## Feature request

**Is your feature request related to a problem? Please describe.**
<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->
I am using Starrocks + Apache Polaris + Azure (ADLS). I would like to set credentials to storage only in Polaris and let it to generate storage credentials for all clients (execution engines) with proper ACLs. 
I can use vended credentials mechanism with ADLS an Polaris in other tools:
* Kafka Connect (IcebergSinkConnector) with config like this:
```
{
"name": "iceberg-test-sink",
"config": {
    "connector.class": "org.apache.iceberg.connect.IcebergSinkConnector",
    "tasks.max": "1",
    "topics": "my_test_topic",
    "iceberg.tables": "test_db.test_table",
    "iceberg.catalog.type": "rest",
    "iceberg.catalog.uri": "http://localhost:8181/api/catalog",
    "iceberg.catalog.credential": "POLARIS_USER:POLARIS_PASSWORD",
    "iceberg.catalog.scope": "PRINCIPAL_ROLE:ALL",
    "iceberg.catalog.warehouse": "polaris",
    "iceberg.catalog.io-impl": "org.apache.iceberg.azure.adlsv2.ADLSFileIO",
    "iceberg.catalog.include-credentials": "true",
    "iceberg.catalog.header.X-Iceberg-Access-Delegation": "vended-credentials",
    "iceberg.control.commit.interval-ms": 50000
    }
}
```
* Spark:
```
pyspark \
--packages org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:1.8.1,org.apache.iceberg:iceberg-azure-bundle:1.8.1,org.apache.hadoop:hadoop-azure:3.4.1,org.apache.hadoop:hadoop-azure-datalake:3.4.1 \
--conf spark.sql.extensions=org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions \
--conf spark.sql.catalog.polaris.warehouse=polaris \
--conf spark.sql.catalog.polaris.header.X-Iceberg-Access-Delegation=vended-credentials \
--conf spark.sql.catalog.polaris=org.apache.iceberg.spark.SparkCatalog \
--conf spark.sql.catalog.polaris.catalog-impl=org.apache.iceberg.rest.RESTCatalog \
--conf spark.sql.catalog.polaris.uri=http://polaris:8181/api/catalog \
--conf spark.sql.catalog.polaris.credential=POLARIS_USER:POLARIS_PASSWORD \
--conf spark.sql.catalog.polaris.scope=PRINCIPAL_ROLE:ALL \
--conf spark.sql.catalog.polaris.token-refresh-enabled=true
```
But in Starrocks doc I can see that I have to set Azure credentials at Starrocks side: https://docs.starrocks.io/docs/data_source/catalog/iceberg/iceberg_catalog/#storagecredentialparams

So my current catalog definition in Starrocks looks like this:
```
CREATE EXTERNAL CATALOG 'iceberg'
COMMENT "External catalog to Apache Iceberg"
PROPERTIES
(
  "iceberg.catalog.uri" = "http://polaris:8181/api/catalog", 
  "type"="iceberg",
  "iceberg.catalog.type"="rest",
  "iceberg.catalog.warehouse"="polaris",
  "iceberg.catalog.security" = "oauth2",
  "iceberg.catalog.oauth2.credential" = "POLARIS_USER:POLARIS_PASSWORD,
  "iceberg.catalog.oauth2.scope"='PRINCIPAL_ROLE:ALL',
  "azure.adls2.storage_account" = "XXX",
  "azure.adls2.shared_key" = "XXX"
);
```

**Describe the solution you'd like**
<!-- A clear and concise description of what you want to happen. -->
I would like to be able to define and use (read/write data) catalog with Azure credentials vended from iceberg rest catalog, e.g. Polaris. E.g.:
```
CREATE EXTERNAL CATALOG 'iceberg'
COMMENT "External catalog to Apache Iceberg"
PROPERTIES
(
  "iceberg.catalog.uri" = "http://polaris:8181/api/catalog", 
  "type"="iceberg",
  "iceberg.catalog.type"="rest",
  "iceberg.catalog.warehouse"="polaris",
  "iceberg.catalog.security" = "oauth2",
  "iceberg.catalog.oauth2.credential" = "POLARIS_USER:POLARIS_PASSWORD,
  "iceberg.catalog.oauth2.scope"='PRINCIPAL_ROLE:ALL'
);
set catalog iceberg;
create database test_db;
create table iceberg.test_db.test_table(
	`id` VARCHAR(50) comment "record id",
	`login` VARCHAR(100) comment "some login"
)
comment "some test table";
insert into iceberg.test_db.test_table values ("a", "b");
```
Now I am getting:
```
SQL Error [5025] [HY000]: fail to connect hdfs namenode, namenode=abfss://polaris@XXX.dfs.core.windows.net/, err=error=Unknown error 255, root_cause=KeyProviderException: Failure to initialize configuration for XXX.dfs.core.windows.net key ="null": Invalid configuration value detected for fs.azure.account.key: BE:10001
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:130)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:654)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$2(SQLQueryJob.java:545)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:567)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:1039)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:4301)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:128)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:192)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:126)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:5026)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:119)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.sql.SQLException: fail to connect hdfs namenode, namenode=abfss://polaris@XXX.dfs.core.windows.net/, err=error=Unknown error 255, root_cause=KeyProviderException: Failure to initialize configuration for XXX.dfs.core.windows.net key ="null": Invalid configuration value detected for fs.azure.account.key: BE:10001
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:130)
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
	at com.mysql.cj.jdbc.StatementImpl.executeInternal(StatementImpl.java:770)
	at com.mysql.cj.jdbc.StatementImpl.execute(StatementImpl.java:653)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:344)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:128)
```

**Describe alternatives you've considered**
<!-- A clear and concise description of any alternative solutions or features you've considered. -->
I have tried to set some params I have found in the internet, but nothing was working.

**Additional context**
<!-- Add any other context or screenshots about the feature request here. -->
I was testing latest versions of components:
* Starrocks 3.4.0
* Polaris 0.9.0 (image built from main branch of github repo)
* Spark 3.5.5

--- a/fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java
+++ b/fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java
@@ -52,6 +52,9 @@ import com.github.benmanes.caffeine.cache.Cache;
 import com.github.benmanes.caffeine.cache.Caffeine;
 import com.github.benmanes.caffeine.cache.Weigher;
 import com.starrocks.common.Config;
+import com.starrocks.common.StarRocksException;
+import com.starrocks.connector.exception.StarRocksConnectorException;
+import com.starrocks.fs.azure.AzBlobURI;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.LocatedFileStatus;
@@ -89,6 +92,10 @@ import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 import java.util.regex.Pattern;
 
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.ADLS_ENDPOINT;
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.ADLS_SAS_TOKEN;
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.BLOB_ENDPOINT;
+
 /**
  * Implementation of FileIO that adds metadata content caching features.
  */
@@ -106,16 +113,16 @@ public class IcebergCachingFileIO implements FileIO, HadoopConfigurable {
     public static final long DISK_CACHE_EXPIRATION_SECONDS = Config.iceberg_metadata_disk_cache_expiration_seconds;
 
     private transient ContentCache fileContentCache;
-    private FileIO wrappedIO;
+    private ResolvingFileIO wrappedIO;
+    private Map<String, String> properties;
     private SerializableSupplier<Configuration> conf;
     private static final Pattern HADOOP_CATALOG_METADATA_JSON_PATTERN =
             Pattern.compile("^v\\d+(\\.gz)?\\.metadata\\.json(\\.gz)?$");
 
     @Override
     public void initialize(Map<String, String> properties) {
-        ResolvingFileIO resolvingFileIO = new ResolvingFileIO();
-        resolvingFileIO.setConf(conf.get());
-        wrappedIO = resolvingFileIO;
+        this.properties = properties;
+        wrappedIO = new ResolvingFileIO();
         wrappedIO.initialize(properties);
 
         if (ENABLE_DISK_CACHE) {
@@ -158,12 +165,26 @@ public class IcebergCachingFileIO implements FileIO, HadoopConfigurable {
 
     @Override
     public InputFile newInputFile(String path) {
-        return new CachingInputFile(fileContentCache, wrappedIO.newInputFile(path));
+        try {
+            wrappedIO.setConf(buildConfFromProperties(properties, path));
+            return new CachingInputFile(fileContentCache, wrappedIO.newInputFile(path));
+        } catch (StarRocksException e) {
+            String errorMessage = String.format("Failed to new input file for path: %s, properties: %s", path, properties);
+            LOG.error(errorMessage, e);
+            throw new StarRocksConnectorException(errorMessage, e);
+        }
     }
 
     @Override
     public OutputFile newOutputFile(String path) {
-        return wrappedIO.newOutputFile(path);
+        try {
+            wrappedIO.setConf(buildConfFromProperties(properties, path));
+            return wrappedIO.newOutputFile(path);
+        } catch (StarRocksException e) {
+            String errorMessage = String.format("Failed to new output file for path: %s, properties: %s", path, properties);
+            LOG.error(errorMessage, e);
+            throw new StarRocksConnectorException(errorMessage, e);
+        }
     }
 
     @Override
@@ -180,6 +201,29 @@ public class IcebergCachingFileIO implements FileIO, HadoopConfigurable {
     @Override
     public Map<String, String> properties() {
         return wrappedIO.properties();
+    }
+
+    public Configuration buildConfFromProperties(Map<String, String> properties, String path) throws StarRocksException {
+        // build Hadoop configuration from properties for HadoopFileIO
+        Configuration copied = new Configuration(conf.get());
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            // Handle Azure ADLS SAS token
+            if (entry.getKey().startsWith(ADLS_SAS_TOKEN) && entry.getKey().endsWith(ADLS_ENDPOINT)) {
+                String endpoint = entry.getKey().substring(ADLS_SAS_TOKEN.length());
+                copied.set(String.format("fs.azure.account.auth.type.%s", endpoint),
+                        "SAS");
+                copied.set(String.format("fs.azure.sas.fixed.token.%s", endpoint),
+                        entry.getValue());
+                return copied;
+            } else if (entry.getKey().startsWith(ADLS_SAS_TOKEN) && (entry.getKey().endsWith(BLOB_ENDPOINT))) {
+                AzBlobURI uri = AzBlobURI.parse(path);
+                String key =
+                        String.format("fs.azure.sas.%s.%s.blob.core.windows.net", uri.getContainer(), uri.getAccount());
+                copied.set(key, entry.getValue());
+                return copied;
+            }
+        }
+        return conf.get();
     }
 
     private static class CacheEntry {

--- a/fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationFactory.java
+++ b/fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationFactory.java
@@ -15,6 +15,7 @@
 package com.starrocks.credential;
 
 import com.google.common.collect.ImmutableList;
+import com.starrocks.common.StarRocksException;
 import com.starrocks.connector.share.credential.CloudConfigurationConstants;
 import com.starrocks.credential.aliyun.AliyunCloudConfigurationProvider;
 import com.starrocks.credential.aws.AwsCloudConfigurationProvider;
@@ -24,15 +25,22 @@ import com.starrocks.credential.gcp.GCPCloudConfigurationProvoder;
 import com.starrocks.credential.hdfs.HDFSCloudConfigurationProvider;
 import com.starrocks.credential.hdfs.StrictHDFSCloudConfigurationProvider;
 import com.starrocks.credential.tencent.TencentCloudConfigurationProvider;
+import com.starrocks.fs.azure.AzBlobURI;
 import org.apache.hadoop.hive.conf.HiveConf;
 import org.apache.iceberg.aws.AwsClientProperties;
 import org.apache.iceberg.aws.s3.S3FileIOProperties;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 import java.util.HashMap;
 import java.util.Map;
 
-public class CloudConfigurationFactory {
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.ADLS_ENDPOINT;
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.ADLS_SAS_TOKEN;
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.BLOB_ENDPOINT;
 
+public class CloudConfigurationFactory {
+    private static final Logger LOG = LogManager.getLogger(CloudConfigurationFactory.class);
     static ImmutableList<CloudConfigurationProvider> cloudConfigurationFactoryChain = ImmutableList.of(
             new AwsCloudConfigurationProvider(),
             new AzureCloudConfigurationProvider(),
@@ -83,7 +91,18 @@ public class CloudConfigurationFactory {
         return null;
     }
 
-    public static CloudConfiguration buildCloudConfigurationForVendedCredentials(Map<String, String> properties) {
+    public static CloudConfiguration buildCloudConfigurationForVendedCredentials(Map<String, String> properties,
+                                                                                 String path) {
+        CloudConfiguration cloudConfiguration = buildCloudConfigurationForAWSVendedCredentials(properties);
+        if (cloudConfiguration.getCloudType() != CloudType.DEFAULT) {
+            return cloudConfiguration;
+        }
+
+        cloudConfiguration = buildCloudConfigurationForAzureVendedCredentials(properties, path);
+        return cloudConfiguration;
+    }
+
+    public static CloudConfiguration buildCloudConfigurationForAWSVendedCredentials(Map<String, String> properties) {
         Map<String, String> copiedProperties = new HashMap<>();
         String sessionAk = properties.getOrDefault(S3FileIOProperties.ACCESS_KEY_ID, null);
         String sessionSk = properties.getOrDefault(S3FileIOProperties.SECRET_ACCESS_KEY, null);
@@ -106,6 +125,30 @@ public class CloudConfigurationFactory {
             }
             if (enablePathStyle != null) {
                 copiedProperties.put(CloudConfigurationConstants.AWS_S3_ENABLE_PATH_STYLE_ACCESS, enablePathStyle);
+            }
+        }
+        return buildCloudConfigurationForStorage(copiedProperties);
+    }
+
+    public static CloudConfiguration buildCloudConfigurationForAzureVendedCredentials(Map<String, String> properties,
+                                                                                      String path) {
+        Map<String, String> copiedProperties = new HashMap<>();
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            if (entry.getKey().startsWith(ADLS_SAS_TOKEN) && entry.getKey().endsWith(ADLS_ENDPOINT)) {
+                String endpoint = entry.getKey().substring(ADLS_SAS_TOKEN.length());
+                copiedProperties.put(CloudConfigurationConstants.AZURE_ADLS2_ENDPOINT, endpoint);
+                copiedProperties.put(CloudConfigurationConstants.AZURE_ADLS2_SAS_TOKEN, entry.getValue());
+            } else if (entry.getKey().startsWith(ADLS_SAS_TOKEN) && entry.getKey().endsWith(BLOB_ENDPOINT)) {
+                try {
+                    AzBlobURI uri = AzBlobURI.parse(path);
+                    copiedProperties.put(CloudConfigurationConstants.AZURE_BLOB_STORAGE_ACCOUNT, uri.getAccount());
+                    copiedProperties.put(CloudConfigurationConstants.AZURE_BLOB_CONTAINER, uri.getContainer());
+                    copiedProperties.put(CloudConfigurationConstants.AZURE_BLOB_SAS_TOKEN, entry.getValue());
+                } catch (StarRocksException e) {
+                    String errorMessage = String.format("Failed to parse azure blob file for path: %s, properties: %s", path,
+                            properties);
+                    LOG.error(errorMessage, e);
+                }
             }
         }
         return buildCloudConfigurationForStorage(copiedProperties);

--- a/fe/fe-core/src/main/java/com/starrocks/credential/azure/AzureCloudConfigurationProvider.java
+++ b/fe/fe-core/src/main/java/com/starrocks/credential/azure/AzureCloudConfigurationProvider.java
@@ -49,6 +49,10 @@ public class AzureCloudConfigurationProvider implements CloudConfigurationProvid
     // Used to retrieve azure load path from configuration map
     public static final String AZURE_PATH_KEY = "azure_path_key";
 
+    public static final String ADLS_ENDPOINT = "dfs.core.windows.net";
+    public static final String BLOB_ENDPOINT = "blob.core.windows.net";
+    public static final String ADLS_SAS_TOKEN = "adls.sas-token.";
+
     @Override
     public CloudConfiguration build(Map<String, String> properties) {
         Preconditions.checkNotNull(properties);

--- a/fe/fe-core/src/main/java/com/starrocks/credential/azure/AzureStorageCloudCredential.java
+++ b/fe/fe-core/src/main/java/com/starrocks/credential/azure/AzureStorageCloudCredential.java
@@ -324,8 +324,10 @@ class AzureADLS2CloudCredential extends AzureStorageCloudCredential {
                 "oauth2ManagedIdentity=" + oauth2ManagedIdentity +
                 ", oauth2TenantId='" + oauth2TenantId + '\'' +
                 ", oauth2ClientId='" + oauth2ClientId + '\'' +
+                ", endpoint='" + endpoint + '\'' +
                 ", storageAccount='" + storageAccount + '\'' +
                 ", sharedKey='" + sharedKey + '\'' +
+                ", sasToken='" + sasToken + '\'' +
                 ", oauth2ClientSecret='" + oauth2ClientSecret + '\'' +
                 ", oauth2ClientEndpoint='" + oauth2ClientEndpoint + '\'' +
                 '}';

--- a/fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java
+++ b/fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java
@@ -148,7 +148,8 @@ public class IcebergScanNode extends ScanNode {
         // Hard coding here
         // Try to get tabular signed temporary credential
         CloudConfiguration vendedCredentialsCloudConfiguration = CloudConfigurationFactory.
-                buildCloudConfigurationForVendedCredentials(icebergTable.getNativeTable().io().properties());
+                buildCloudConfigurationForVendedCredentials(icebergTable.getNativeTable().io().properties(),
+                        icebergTable.getNativeTable().location());
         if (vendedCredentialsCloudConfiguration.getCloudType() != CloudType.DEFAULT) {
             // If we get CloudConfiguration succeed from iceberg FileIO's properties, we just using it.
             cloudConfiguration = vendedCredentialsCloudConfiguration;

--- a/fe/fe-core/src/main/java/com/starrocks/planner/IcebergTableSink.java
+++ b/fe/fe-core/src/main/java/com/starrocks/planner/IcebergTableSink.java
@@ -69,7 +69,8 @@ public class IcebergTableSink extends DataSink {
 
         // Try to set for tabular
         CloudConfiguration tabularTempCloudConfiguration = CloudConfigurationFactory.
-                buildCloudConfigurationForVendedCredentials(icebergTable.getNativeTable().io().properties());
+                buildCloudConfigurationForVendedCredentials(icebergTable.getNativeTable().io().properties(),
+                        this.location);
         if (tabularTempCloudConfiguration.getCloudType() != CloudType.DEFAULT) {
             this.cloudConfiguration = tabularTempCloudConfiguration;
         } else {

--- a/fe/fe-core/src/test/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIOTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIOTest.java
@@ -15,6 +15,7 @@
 package com.starrocks.connector.iceberg.io;
 
 import com.starrocks.common.StarRocksException;
+import com.starrocks.connector.exception.StarRocksConnectorException;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.iceberg.io.InputFile;
 import org.junit.jupiter.api.Assertions;
@@ -63,6 +64,25 @@ public class IcebergCachingFileIOTest {
         long cacheIOInputFileSize = cachingFileIOInputFile.getLength();
         Assertions.assertEquals(cacheIOInputFileSize, 39);
         cachingFileIO.deleteFile(path);
+    }
+
+    @Test
+    public void testNewFileWithException() {
+        IcebergCachingFileIO cachingFileIO = new IcebergCachingFileIO();
+        cachingFileIO.setConf(new Configuration());
+        Map<String, String> icebergProperties = new HashMap<>();
+        String key = ADLS_SAS_TOKEN + "account." + BLOB_ENDPOINT;
+        icebergProperties.put(key, "sas_token");
+        cachingFileIO.initialize(icebergProperties);
+
+        String path = "file:/tmp/non_existent_file.json";
+        Assertions.assertThrows(StarRocksConnectorException.class, () -> {
+            cachingFileIO.newInputFile(path);
+        });
+
+        Assertions.assertThrows(StarRocksConnectorException.class, () -> {
+            cachingFileIO.newOutputFile(path);
+        });
     }
 
     @Test

--- a/fe/fe-core/src/test/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIOTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIOTest.java
@@ -14,6 +14,7 @@
 
 package com.starrocks.connector.iceberg.io;
 
+import com.starrocks.common.StarRocksException;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.iceberg.io.InputFile;
 import org.junit.jupiter.api.Assertions;
@@ -24,6 +25,10 @@ import java.io.FileWriter;
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.ADLS_ENDPOINT;
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.ADLS_SAS_TOKEN;
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.BLOB_ENDPOINT;
 
 public class IcebergCachingFileIOTest {
 
@@ -58,5 +63,36 @@ public class IcebergCachingFileIOTest {
         long cacheIOInputFileSize = cachingFileIOInputFile.getLength();
         Assertions.assertEquals(cacheIOInputFileSize, 39);
         cachingFileIO.deleteFile(path);
+    }
+
+    @Test
+    public void testBuildConfFromProperties() throws StarRocksException {
+        Map<String, String> properties = new HashMap<>();
+        String key = ADLS_SAS_TOKEN + "account." + ADLS_ENDPOINT;
+        String sasToken = "sas_token";
+        properties.put(key, sasToken);
+        String path = "abfss://container@account.dfs.core.windows.net/path/1/2";
+
+        IcebergCachingFileIO cachingFileIO = new IcebergCachingFileIO();
+        cachingFileIO.setConf(new Configuration());
+        Configuration configuration = cachingFileIO.buildConfFromProperties(properties, path);
+
+        String authType = configuration.get("fs.azure.account.auth.type.account." + ADLS_ENDPOINT);
+        Assertions.assertEquals("SAS", authType);
+        String token = configuration.get("fs.azure.sas.fixed.token.account." + ADLS_ENDPOINT);
+        Assertions.assertEquals(sasToken, token);
+
+        properties = new HashMap<>();
+        key = ADLS_SAS_TOKEN + "account." + BLOB_ENDPOINT;
+        sasToken = "blob_sas_token";
+        properties.put(key, sasToken);
+        path = "wasbs://container@account.blob.core.windows.net/path/1/2";
+
+        cachingFileIO = new IcebergCachingFileIO();
+        cachingFileIO.setConf(new Configuration());
+        configuration = cachingFileIO.buildConfFromProperties(properties, path);
+
+        token = configuration.get("fs.azure.sas.container.account." + BLOB_ENDPOINT);
+        Assertions.assertEquals(sasToken, token);
     }
 }

--- a/fe/fe-core/src/test/java/com/starrocks/credential/CloudConfigurationFactoryTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/credential/CloudConfigurationFactoryTest.java
@@ -291,8 +291,8 @@ public class CloudConfigurationFactoryTest {
         cc.toFileStoreInfo();
         Assertions.assertEquals(cc.toConfString(),
                 "AzureCloudConfiguration{resources='', jars='', hdpuser='', " +
-                        "cred=AzureADLS2CloudCredential{oauth2ManagedIdentity=false, " +
-                        "oauth2TenantId='XX', oauth2ClientId='XX', storageAccount='XX', sharedKey='XX', " +
+                        "cred=AzureADLS2CloudCredential{oauth2ManagedIdentity=false, oauth2TenantId='XX', " +
+                        "oauth2ClientId='XX', endpoint='', storageAccount='XX', sharedKey='XX', sasToken='', " +
                         "oauth2ClientSecret='XX', oauth2ClientEndpoint='XX'}}");
     }
 

--- a/fe/fe-core/src/test/java/com/starrocks/credential/CloudConfigurationFactoryTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/credential/CloudConfigurationFactoryTest.java
@@ -28,17 +28,22 @@ import org.junit.jupiter.api.Test;
 import java.util.HashMap;
 import java.util.Map;
 
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.ADLS_ENDPOINT;
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.ADLS_SAS_TOKEN;
+import static com.starrocks.credential.azure.AzureCloudConfigurationProvider.BLOB_ENDPOINT;
+
 public class CloudConfigurationFactoryTest {
 
     @Test
-    public void testBuildCloudConfigurationForVendedCredentials() {
+    public void testBuildCloudConfigurationForAWSVendedCredentials() {
         Map<String, String> map = new HashMap<>();
         map.put(S3FileIOProperties.ACCESS_KEY_ID, "ak");
         map.put(S3FileIOProperties.SECRET_ACCESS_KEY, "sk");
         map.put(S3FileIOProperties.SESSION_TOKEN, "token");
         map.put(S3FileIOProperties.PATH_STYLE_ACCESS, "true");
         map.put(AwsClientProperties.CLIENT_REGION, "region");
-        CloudConfiguration cloudConfiguration = CloudConfigurationFactory.buildCloudConfigurationForVendedCredentials(map);
+        CloudConfiguration cloudConfiguration = CloudConfigurationFactory.buildCloudConfigurationForVendedCredentials(map,
+                "");
         Assertions.assertNotNull(cloudConfiguration);
         Assertions.assertEquals(CloudType.AWS, cloudConfiguration.getCloudType());
         Assertions.assertEquals(
@@ -53,7 +58,7 @@ public class CloudConfigurationFactoryTest {
         map.remove(S3FileIOProperties.PATH_STYLE_ACCESS);
         map.put(S3FileIOProperties.ENDPOINT, "endpoint");
         map.put(CloudConfigurationConstants.AWS_S3_REGION, "us-west-2");
-        cloudConfiguration = CloudConfigurationFactory.buildCloudConfigurationForVendedCredentials(map);
+        cloudConfiguration = CloudConfigurationFactory.buildCloudConfigurationForVendedCredentials(map, "");
         Assertions.assertNotNull(cloudConfiguration);
         Assertions.assertEquals(CloudType.AWS, cloudConfiguration.getCloudType());
         Assertions.assertEquals(
@@ -62,6 +67,35 @@ public class CloudConfigurationFactoryTest {
                         "useInstanceProfile=false, accessKey='ak', secretKey='sk', " +
                         "sessionToken='token', iamRoleArn='', stsRegion='', stsEndpoint='', externalId='', " +
                         "region='us-west-2', endpoint='endpoint'}, enablePathStyleAccess=false, enableSSL=true}",
+                cloudConfiguration.toConfString());
+    }
+
+    @Test
+    public void testBuildCloudConfigurationForAzureVendedCredentials() {
+        Map<String, String> map = new HashMap<>();
+        map.put(ADLS_SAS_TOKEN + "account." + ADLS_ENDPOINT, "sas_token");
+        CloudConfiguration cloudConfiguration = CloudConfigurationFactory.buildCloudConfigurationForVendedCredentials(map,
+                "abfss://container@account.dfs.core.windows.net/path/1/2");
+        Assertions.assertNotNull(cloudConfiguration);
+        Assertions.assertEquals(CloudType.AZURE, cloudConfiguration.getCloudType());
+        Assertions.assertEquals(
+                "AzureCloudConfiguration{resources='', jars='', hdpuser='', " +
+                        "cred=AzureADLS2CloudCredential{oauth2ManagedIdentity=false, oauth2TenantId='', oauth2ClientId='', " +
+                        "endpoint='account.dfs.core.windows.net', storageAccount='', sharedKey='', " +
+                        "sasToken='sas_token', oauth2ClientSecret='', oauth2ClientEndpoint=''}}",
+                cloudConfiguration.toConfString());
+
+        map = new HashMap<>();
+        map.put(ADLS_SAS_TOKEN + "account." + BLOB_ENDPOINT, "sas_token");
+        cloudConfiguration = CloudConfigurationFactory.buildCloudConfigurationForVendedCredentials(map,
+                "wasbs://container@account.blob.core.windows.net/path/1/2");
+        Assertions.assertNotNull(cloudConfiguration);
+        Assertions.assertEquals(CloudType.AZURE, cloudConfiguration.getCloudType());
+        Assertions.assertEquals(
+                "AzureCloudConfiguration{resources='', jars='', hdpuser='', " +
+                        "cred=AzureBlobCloudCredential{endpoint='', storageAccount='account', sharedKey='', " +
+                        "container='container', sasToken='sas_token', useManagedIdentity='false', clientId='', " +
+                        "clientSecret='', tenantId=''}}",
                 cloudConfiguration.toConfString());
     }
 

